New Settings Dialog

Schneegans opened the pull request:
This is still work in progress. But you can already test this feature by downloading the latest build from the [latest test builds](https://github.com/kando-menu/kando/actions/workflows/test_deploy.yml) for this branch.

## Backup your Settings!

**Before you test this branch, you should make a backup of your [Kando configuration files](https://kando.menu/config-files/#location-of-the-configuration-files)!** They should still be mostly compatible, but you never know. Jumping between different versions is definitely a bit dangerous and you may loose something in the process. Also please report any issues you find in our [Discord server](https://discord.gg/hZwbVSDkhy)!

## Missing Features

- [ ] Add support for translations
- [x] Add a introduction dialog
- [x] Add better / more self-explanatory icons for the menu items
- [x] Split the open-dev-tools button into two: One for the settings window and one for the menu window
- [x] Add some widgets to the general settings dialog to choose the sound theme
- [x] Allow drag and drop from menu items to other menus

## Known Issues

If you find any issues, please drop a comment below. Then I can add it to this list of things which need to be done before merging this.

- [x] Dropdowns have a weird background color in dark mode
- [x] Some fixed-angle configurations lead to hidden items when hovering
- [x] Menus cannot be unbound
- [x] Test and potentially fix touch input
- [x] When dragging an item in the menu preview, the selected item sometimes changes
- [x] On macOS, the menu and the settings dialog are opened and closed simultaneously which makes this hardly usable
- [x] Centered mode does not work if the menu is scaled
- [x] Some builtin menu themes look weird

yar2000T commented:
I tested the whole application, no new issues were found, and also fixed bugs with the centered mode working correctly! Nice work!

Schneegans commented:
> I tested the whole application, no new issues were found, and also fixed bugs with the centered mode working correctly! Nice work!

Thanks for the feedback! I have also added a default sound theme so that users can at least test the feature :smile:
